contracts/core/RebalancingSetToken.sol

```diff
@@ -534,15 +584,61 @@ contract RebalancingSetToken is
                 combinedCurrentUnits.push(uint256(0));
             }
 
-            // Compute and push unit amounts of token in rebalancingSet, push 0 if not in set
+            // Compute and push unit amounts of token in nextSet, push 0 if not in set
             if (isInRebalance) {
-                combinedRebalanceUnits.push(
-                    computeUnits(rebalancingSetUnits[indexRebalance], rebalancingSetNaturalUnit)
+                combinedNextSetUnits.push(
+                    computeUnits(nextSetUnits[indexRebalance], nextSetNaturalUnit)
                 );
             } else {
-                combinedRebalanceUnits.push(uint256(0));
+                combinedNextSetUnits.push(uint256(0));
+            }
+        }
+    }
+
+    /**
+     * Calculate the amount of nextSets to issue by using the component amounts in the
+     * vault, unitShares following from this calculation.
+     * @return uint256              Amount of nextSets to issue
+     * @return uint256              New unitShares for the rebalancingSetToken
+     */
+    function calculateNextSetIssueQuantity()
+        internal
+        returns (uint256, uint256)
+    {
+        // Collect data necessary to compute issueAmounts
+        uint256 nextNaturalUnit = ISetToken(nextSet).naturalUnit();
+        address[] memory nextComponents = ISetToken(nextSet).getComponents();
+        uint256[] memory nextUnits = ISetToken(nextSet).getUnits();
+        uint256 maxIssueAmount = CommonMath.maxUInt256();
+
+        // Set up vault interface
+        address vaultAddress = ICore(ISetFactory(factory).core()).vault();
+        IVault vault = IVault(vaultAddress);
+
+        for (uint256 i = 0; i < nextComponents.length; i++) {
+            // Get amount of components in vault owned by rebalancingSetToken
+            uint256 componentAmount = vault.getOwnerBalance(
+                nextComponents[i],
+                this
+            );
+
+            // Calculate amount of Sets that can be issued from those components, if less than amount for other
+            // components then set that as maxIssueAmount
+            uint256 componentIssueAmount = componentAmount.div(nextUnits[i]).mul(nextNaturalUnit);
+            if (componentIssueAmount < maxIssueAmount) {
+                maxIssueAmount = componentIssueAmount;
             }
         }
+
+        // Calculate the amount of naturalUnits worth of rebalancingSetToken outstanding
+        uint256 naturalUnitsOutstanding = totalSupply_.div(naturalUnit);
+
+        // Issue amount of Sets that is closest multiple of nextNaturalUnit to the maxIssueAmount
+        uint256 issueAmount = maxIssueAmount.div(nextNaturalUnit).mul(nextNaturalUnit);
```

Comment:
Forgot, where does the dust that is not used to issue remain and who is the owner?

Reply:
Remains in the vault under the rebalancingSetToken's "ownership" so could be used in a future rebalance.